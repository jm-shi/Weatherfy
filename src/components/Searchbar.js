import React from 'react';
import { Link } from 'react-router-dom';

class Searchbar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      zipcode: ''
    };

    this.setZipcode = this.setZipcode.bind(this);
    this.checkForm = this.checkForm.bind(this);
  }

  setZipcode(e) {
    var value = e.target.value;
    this.setState(() => ({ zipcode: value }));
  }

  checkForm(e) {
    if (this.state.zipcode.length !== 5) {
      e.preventDefault();
      alert('Enter a 5-digit zipcode.');
    }
  }

  render() {
    const { zipcode } = this.state;

    return (
      <div className="search-container">
        <input
          id="search-bar"
          type="text"
          placeholder="Enter zipcode"
          value={zipcode}
          onChange={this.setZipcode}
        />

        <Link
          className="button"
          to={{
            pathname: `/forecast/${zipcode}`,
            zipcode: zipcode
          }}
          onClick={this.checkForm}
        >
          Search
        </Link>
      </div>
    );
  }
}

module.exports = Searchbar;
