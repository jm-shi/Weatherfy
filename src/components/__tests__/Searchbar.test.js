import React from 'react';
import Searchbar from '../Searchbar.js';
import { shallow } from 'enzyme';

test('should render Searchbar correctly', () => {
  const wrapper = shallow(<Searchbar />);
  expect(wrapper).toMatchSnapshot();
});
