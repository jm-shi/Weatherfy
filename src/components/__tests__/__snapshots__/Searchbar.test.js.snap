// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Searchbar correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Searchbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          id="search-bar"
          onChange={[Function]}
          placeholder="Enter zipcode"
          type="text"
          value=""
        />,
        <Link
          className="button"
          onClick={[Function]}
          replace={false}
          to={
            Object {
              "pathname": "/forecast/",
              "zipcode": "",
            }
          }
        >
          Search
        </Link>,
      ],
      "className": "search-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "search-bar",
          "onChange": [Function],
          "placeholder": "Enter zipcode",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Search",
          "className": "button",
          "onClick": [Function],
          "replace": false,
          "to": Object {
            "pathname": "/forecast/",
            "zipcode": "",
          },
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="search-bar"
            onChange={[Function]}
            placeholder="Enter zipcode"
            type="text"
            value=""
          />,
          <Link
            className="button"
            onClick={[Function]}
            replace={false}
            to={
              Object {
                "pathname": "/forecast/",
                "zipcode": "",
              }
            }
          >
            Search
          </Link>,
        ],
        "className": "search-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "search-bar",
            "onChange": [Function],
            "placeholder": "Enter zipcode",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Search",
            "className": "button",
            "onClick": [Function],
            "replace": false,
            "to": Object {
              "pathname": "/forecast/",
              "zipcode": "",
            },
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
